mq.go

```diff
@@ -104,22 +131,64 @@ func (mq *MQ) setupProducers(producers Producers) error {
 }
 
 func (mq *MQ) registerProducer(config ProducerConfig) error {
-
 	if _, ok := mq.producers.Get(config.Name); ok {
 		return fmt.Errorf(`Producer with name "%s" is already registered`, config.Name)
 	}
 
-	producer := producer{
-		channel:        mq.channel,
-		errorChannel:   mq.internalErrorChannel,
-		exchange:       config.Exchange,
-		options:        wabbit.Option(config.Options),
-		publishChannel: make(chan []byte, config.BufferSize),
-		routingKey:     config.RoutingKey,
-	}
+	producer := newProducer(mq.channel, mq.internalErrorChannel, config)
 
 	go producer.worker()
-	mq.producers.Set(config.Name, &producer)
+	mq.producers.Set(config.Name, producer)
+
+	return nil
+}
+
+func (mq *MQ) setupConsumers(consumers Consumers) error {
+	for _, config := range consumers {
+		if err := mq.registerConsumer(config); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (mq *MQ) registerConsumer(config ConsumerConfig) error {
+	if _, ok := mq.consumers.Get(config.Name); ok {
+		return fmt.Errorf(`Consumer with name "%s" is already registered`, config.Name)
+	}
+
+	// Consumer must have at least one worker.
+	if config.Workers == 0 {
+		config.Workers = 1
+	}
+
+	consumer := newConsumer(config) // We need to save whole config for reconnection.
+
+	for i := 0; i < config.Workers; i++ {
+		deliveries, err := mq.channel.Consume(consumer.queue, "", consumer.options)
+		if err != nil {
+			return fmt.Errorf(`Error during starting consumer %s: %s`, config.Name, err)
```

Comment:
Error during the consumer starting.
or 
Error during the starting of consumer.